src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -708,7 +720,7 @@ void DirectLightingIntegrator::add_non_physical_light_sample_contribution(
     // Add the contribution of this sample to the illumination.
     const float attenuation = light->compute_distance_attenuation(
         m_material_sampler.get_point(), emission_position);
-    const float weight = transmission * attenuation / sample.m_probability;
+    const float weight = transmission * attenuation / sample.m_probability / probability;
```

Comment:
You can avoid one division by dividing by the product of probabilities:

    const float weight = transmission * attenuation / (sample.m_probability * probability);

This is the kind of optimization the compiler probably cannot do because it may slightly change the result (unless "fast math" is enabled, which it isn't in appleseed).